src/color.rs

```diff
@@ -991,6 +1138,51 @@ mod tests {
             ((0.0, 0.0, 0.0), (0.0, 0.0, 0.0)),
         ]
     }
+
+    #[test]
+    fn ensure_color_mix() {
+        let mix_1 = Color::rgba(18., 55., 125., 1.0);
+        let mix_2 = Color::rgba(125., 33., 200., 0.8);
+        let expected = Color::rgba(72., 44., 163., 0.88);
+        let mix_res = mix_1.mix(&mix_2, 40);
+        assert_eq!(expected, mix_res);
+    }
+
+    #[test]
+    #[should_panic(expected = "101 is not a valid value for weight. Must be within 0-100 inclusive")]
+    fn invalid_mix_weight() {
+        let c1 = Color::rgb(1., 1., 1.);
+        let _ = c1.mix(&Color::rgb(2., 2., 2.), 101);
+    }
+
+    #[test]
+    fn check_rgb_to_hsl() {
+        let c = Color::rgb(251., 206., 33.);
+        let result = c.to_hsl();
+
+        assert_eq!((48., 0.9646017699115044, 0.5568627450980392), result);
+    }
+
+    #[test]
+    fn check_color_hue() {
+        let c = Color::rgb(250., 206., 33.);
+        assert_eq!(48., c.hue());
+
+        let d = Color::rgb(38., 128., 208.);
+        assert_eq!(208., d.hue());
+    }
+
+    #[test]
+    fn check_with_hue_positive() {
+        let c = Color::rgb(245.0, 130.0, 48.0);
+        assert_eq!(c.with_hue(70.), Color::rgb(130., 245., 48.));
+    }
+
+    #[test]
+    fn check_with_hue_negative() {
+        let c = Color::rgb(245.0, 130.0, 48.0);
+        assert_eq!(c.with_hue(-70.), Color::rgb(245., 48., 196.));
+    }
```

Comment:
One test I would like to see is a test where we create a color using `Color::hsl` and then use the `hue`, `saturation` and `lightness` methods (all in that same test) to get the values back. The result should be the exact same values originally passed in to `Color::hsl`. Feel free to round to a certain amount of decimal places to avoid floating point inaccuracy issues.